src/common/types.cpp

```diff
@@ -560,13 +564,24 @@ LogicalTypeId TransformStringToLogicalType(const string &str) {
 		return LogicalTypeId::TIMESTAMP_TZ;
 	} else if (lower_str == "timetz") {
 		return LogicalTypeId::TIME_TZ;
+	} else if (lower_str == "json") {
+		return LogicalTypeId::JSON;
+	} else if (lower_str == "null") {
+		return LogicalTypeId::SQLNULL;
```

Comment:
This change now allows users to define the NULL type in e.g. create table statements, alter types, casts, etc. I'm not fundamentally against that, but this change would at least require testing NULL types in those locations that previously would not have allowed the null type to exist. Is this required for this PR, or is there another way of handling this functionality?

Reply:
I've added some tests for this functionality in `test/sql/types/test_null_type.test`. Please let me know if this is sufficient or if more tests are needed.